arangod/Agency/Node.cpp

```diff
@@ -539,8 +539,9 @@ template <> bool Node::handle<ERASE>(VPackSlice const& slice) {
 
     if (this->slice().isArray()) {
       if (haveVal) {
+        VPackSlice valToErase = slice.get("val");
         for (auto const& old : VPackArrayIterator(this->slice())) {
-          if (old != slice.get("val")) {
+          if (VelocyPackHelper::compare(old, valToErase, /*useUTF8*/true) != 0) {
```

Comment:
We are checking for equality only here, not for < or >. Can't we set `useUtf8` to false then?